### PR TITLE
Document optimizer hints

### PR DESCRIPTION
**Pull request by martinmacko47**
Documents https://github.com/knex/knex/issues/4199 and PR https://github.com/knex/knex/pull/4243

### CODE REVIEW

File: components/Sidebar.jsx [hunk 1]
```diff
@@ -138,6 +138,7 @@ export default class Sidebar extends Component {
           <li>– <a href="#Builder-avg">avg</a></li>
           <li>– <a href="#Builder-increment">increment</a></li>
           <li>– <a href="#Builder-decrement">decrement</a></li>
+          <li>– <a href="#Builder-optimizerHint">optimizerHint</a></li>
```
Comment (martinmacko47) [inline components/Sidebar.jsx:141]: I'm not sure about the order where to put the method

File: sections/builder.js [hunk 2]
```diff
@@ -2439,6 +2439,20 @@ export default [
...
+  {
+    type: "method",
+    method: "optimizerHint",
+    example: ".optimizerHint(hint)",
+    description: "Add hints to the query using comment-like syntax `/*+ ... */`. MySQL and Oracle use this syntax for optimizer hints. Also various DB proxies and routers use this syntax to pass hints to alter their behavior. In other dialects the hints are ignored as simple comments.",
+    children: [
+      {
+        type: "runnable",
+        content: `
+          knex('accounts').where('userid', '=', 1).optimizerHint('hint()')
```
Comment (martinmacko47) [inline sections/builder.js:2451]: `runnable` won't work before the change is released, as the scipt cant compile the query on the current knex version

File: sections/builder.js [hunk 3]
```diff
@@ -2439,6 +2439,20 @@ export default [
       }
     ]
   },
+  {
+    type: "method",
+    method: "optimizerHint",
+    example: ".optimizerHint(hint)",
```
Comment (martinmacko47) [inline sections/builder.js:2445]: Do you have a convention how to denote that `hint` may be either a single value or an array of values?

File: components/Sidebar.jsx (hunk 1 above)
Comment (kibertoad) [reply components/Sidebar.jsx:141]: I think the place is fine.

File: sections/builder.js (hunk 3 above)
Comment (kibertoad) [reply sections/builder.js:2445]: Like this:
`.whereIn(column|columns, array|callback|builder)`

File: sections/builder.js (hunk 2 above)
Comment (kibertoad) [reply sections/builder.js:2451]: Can you use a real hint in the example?

Comment (martinmacko47) [reply sections/builder.js]: @kibertoad Added real MySQL hint example 7bc4df5

File: sections/builder.js (hunk 3 above)
Comment (martinmacko47) [reply sections/builder.js]: @kibertoad Ok. Used `hint|hints` c12ab97

**Review by kibertoad — APPROVED** [review]

**Pull request merged**